File: Suhyun/Level1/1월 3주차/제일 작은 수 제거하기/DeleteSmallNumber.java
import java.util.Arrays;
public class DeleteSmallNumber {
    public static void main(String[] args) {
        
        int[] arr = new int[]{4,3,2,1};
        boolean flag=true;  
        //최솟값 구별위해 미리 초기화      
        int min=arr[0];
        //최솟값 위치 index
        int index=0;
        int[] answer;

        //초기화 
        if(arr.length==1){
            //return arr
            answer=new int[]{-1};
            System.out.println(answer[0]);
            
        }else{
            answer = new int[arr.length-1];
        }
        // 최솟값 index추출 
        for(int i=0; i<arr.length; i++){
            if(min>=arr[i]){
                min=arr[i];
                index=i;
            }
        }


        int j=0;
        for(int i=0; i<answer.length; i++){
            // 최솟값 위치 도달시 뛰어넘도록 
            if(i==index&&flag==true){
                i--;
                j++;
                flag=false;
                continue;
            }
            answer[i]=arr[j];
            j++;
        }
        System.out.println(Arrays.toString(answer));
        //return answer;
    }
}

File: Suhyun/Level1/1월 3주차/제일 작은 수 제거하기/DeleteSmallNumber.js
let arr = [4,3,2,1];
let answer=[];

if(arr.length==1){
    answer[0]=-1;
    return answer;
}
// 미리 최솟값 구하기 
const min =Math.min(...arr);
//filter로 걸러서 min빼고 answer 쪽으로 
answer=arr.filter(e => e !== min);
console.log(answer);
//return answer.filter(e => e !== min);

//return answer=arr.filter((e) =>e!==Math.min.apply(null,arr));




